techlibs/lattice/synth_lattice.cc

```diff
@@ -0,0 +1,574 @@
+		log("        generate the synthesis netlist for the specified family.\n");
+		log("        supported values:\n");
+		log("        - ecp5: ECP5\n");
+		log("        - xo2: MachXO2\n");
+		log("        - xo3: MachXO3L/LF\n");
+		log("        - xo3d: MachXO3D\n");
+		//log("        - xo: MachXO (EXPERIMENTAL)\n");
+		//log("        - pm: Platform Manager (EXPERIMENTAL)\n");
+		//log("        - pm2: Platform Manager 2 (EXPERIMENTAL)\n");
+		//log("        - xp: LatticeXP (EXPERIMENTAL)\n");
+		//log("        - xp2: LatticeXP2 (EXPERIMENTAL)\n");
+		//log("        - ecp: LatticeECP/EC (EXPERIMENTAL)\n");
+		//log("        - sm: LatticeSC/M (EXPERIMENTAL)\n");
+		//log("        - ecp2: LatticeECP2/M (EXPERIMENTAL)\n");
+		//log("        - ecp3: LatticeECP3 (EXPERIMENTAL)\n");
+		//log("        - lifmd: LIFMD (EXPERIMENTAL)\n");
+		//log("        - lifmdf: LIFMDF (EXPERIMENTAL)\n");
+		log("\n");
+		log("    -blif <file>\n");
+		log("        write the design to the specified BLIF file. writing of an output file\n");
+		log("        is omitted if this parameter is not specified.\n");
+		log("\n");
+		log("    -edif <file>\n");
+		log("        write the design to the specified EDIF file. writing of an output file\n");
+		log("        is omitted if this parameter is not specified.\n");
+		log("\n");
+		log("    -json <file>\n");
+		log("        write the design to the specified JSON file. writing of an output file\n");
+		log("        is omitted if this parameter is not specified.\n");
+		log("\n");
+		log("    -run <from_label>:<to_label>\n");
+		log("        only run the commands between the labels (see below). an empty\n");
+		log("        from label is synonymous to 'begin', and empty to label is\n");
+		log("        synonymous to the end of the command list.\n");
+		log("\n");
+		log("    -noflatten\n");
+		log("        do not flatten design before synthesis\n");
+		log("\n");
+		log("    -dff\n");
+		log("        run 'abc'/'abc9' with -dff option\n");
+		log("\n");
+		log("    -retime\n");
+		log("        run 'abc' with '-dff -D 1' options\n");
+		log("\n");
+		log("    -noccu2\n");
+		log("        do not use CCU2 cells in output netlist\n");
+		log("\n");
+		log("    -nodffe\n");
+		log("        do not use flipflops with CE in output netlist\n");
+		log("\n");
+		log("    -nobram\n");
+		log("        do not use block RAM cells in output netlist\n");
+		log("\n");
+		log("    -nolutram\n");
+		log("        do not use LUT RAM cells in output netlist\n");
+		log("\n");
+		log("    -nowidelut\n");
+		log("        do not use PFU muxes to implement LUTs larger than LUT4s\n");
+		log("\n");
+		log("    -asyncprld\n");
+		log("        use async PRLD mode to implement ALDFF (EXPERIMENTAL)\n");
+		log("\n");
+		log("    -abc2\n");
+		log("        run two passes of 'abc' for slightly improved logic density\n");
+		log("\n");
+		log("    -abc9\n");
+		log("        use new ABC9 flow (EXPERIMENTAL)\n");
+		log("\n");
+		log("    -vpr\n");
```

Comment:
can we remove this? there's never been any meaningful VPR support for lattice fpgas, and I think VPR has moved on enough that whatever we output would probably be no use anyway

Reply:
sure vpr and blif can be removed it was added to be fully compatible with synth_ecp5

Reply:
sounds good - this is one of the places where I'd rather see an easy opportunity for cleanup taken rather than 100% backwards compatibility